frontend/src/book-read/components/shared/review-mutation.jsx

```diff
@@ -0,0 +1,138 @@
+import {
+  //   Box,
+  Paper,
+  Rating,
+  TextField,
+  Stack,
+  Button,
+  Typography,
+  Container,
+  Alert,
+} from "@mui/material";
+import { useState, useEffect } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+
+const initialFormState = {
+  score: 0,
+  title: "",
+  text: "",
+};
+
+export default function ReviewMutationComponent({
+  onSubmit,
+  //   onDelete,
+  //   headerTitle,
+  mutationTitle,
+  error,
+  review,
+  //   isLoading,
+  isSuccess,
+  isError,
+}) {
+  const [formState, setFormState] = useState(initialFormState);
+  const { bookId } = useParams();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (isSuccess) {
+      navigate(`/read/books/${bookId}`);
+    }
+  }, [isSuccess, navigate]);
+
+  if (review) {
+    const { score, title, text } = review;
+    useEffect(() => {
+      setFormState({
+        score,
+        title,
+        text,
+      });
+    }, [score, title, text]);
+  }
+
+  function handleRatingChange(event, value) {
```

Comment:
Please, change functions declarations inside components to const variables with arrow functions. This should be done to keep consistency with the approach in the rest of the project.

Reply:
Done